File: client/src/app/service/user/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs/Observable";

import { User } from "../../model/user";


@Injectable()
export class UserService {

  private originUrl = '//localhost:8080/users';

  constructor(private http: HttpClient) { }

  public get(id: number): Observable<User> {
    return this.http.get<User>(`${this.originUrl}/${id}`);
  }

  public getAll(): Observable<User[]> {
    return null;
  }

  public create(user: User): Observable<User> {
    return null;
  }

  public update(user: User): Observable<User> {
    return null;
  }

  public delete(user: User):  Observable<User> {
    return null;
  }

  public restore(user: User): Observable<User> {
    return null;
  }

}

File: server/src/main/java/com/cafetamine/surveys/service/UserService.java
package com.cafetamine.surveys.service;

import org.springframework.stereotype.Service;

import java.util.Optional;

import com.cafetamine.surveys.model.User;
import com.cafetamine.surveys.persistence.UserRepository;


@Service
public class UserService {

    private UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    private User getByIdAndActive(Long id, Boolean isActive) {
        Optional<User> user = this.userRepository.findByIdAndIsActive(id, isActive);
        if (!user.isPresent()) {
            // TODO exception
            throw new RuntimeException("404 user not found");
        }
        return user.get();
    }

    public User getById(Long id) {
        return this.getByIdAndActive(id, true);
    }

    public Iterable<User> getAll() {
        return this.userRepository.findAllByIsActive(true);
    }

    public User create(User user) {
        // TODO adjust implementation to client
        user.setActive(true);
        return this.userRepository.save(user);
    }

    public User update(Long id, User updatedUser) {
        User user = this.getById(id);

        if (!updatedUser.getLogin().equals("")) {
            user.setLogin(updatedUser.getLogin());
        }
        if (!updatedUser.getEmail().equals("")) {
            user.setEmail(updatedUser.getEmail());
        }
        if (!updatedUser.getPassword().equals("")) {
            user.setPassword(updatedUser.getPassword());
        }
        if (!updatedUser.getName().equals("")) {
            user.setName(updatedUser.getName());
        }
        if (!updatedUser.getSurname().equals("")) {
            user.setSurname(updatedUser.getSurname());
        }

        return this.userRepository.save(user);
    }

    public User delete(Long id) {
        User user = this.getById(id);
        user.setActive(false);

        return this.userRepository.save(user);
    }

    public User restore(Long id) {
        User user = this.getByIdAndActive(id, false);
        user.setActive(true);

        return this.userRepository.save(user);
    }

}
